### DISCUSSION TITLE
Referrencing imports

### DISCUSSION THREAD

**Discussion by Babailan** [vercel/next.js#43123] [Help] [answered]
when i import the example.js in api it's making a new initialization.

![image](https://user-images.githubusercontent.com/83863770/202846412-f459fa1d-c7c9-4bf6-a199-a910b90589cd.png)

when i request in the api it make the first initialization how do i reference without making a new initialization.
![image](https://user-images.githubusercontent.com/83863770/202846460-56f2477e-9266-485f-92cf-29cdf55c0424.png)

**Comment by Babailan** [comment]
when i request it's making a new initialization
![image](https://user-images.githubusercontent.com/83863770/202846523-66ee9e32-2c8a-48e6-8e17-2309d2d019d0.png)

**Comment by maxeth** [comment, marked as answer]
Next.js uses serverless API routes that get started and shut down on pretty much a per-request-basis. So after each request, every (imported) code is re-initialized/executed. This is an important aspect of serverless architecture as there is no ability to store state on the "server" as you'd like to do with the constant ```adminApp```. (This is why there are so many popular serverless ("stateless") solutions like Upstash, FaunaDB etc. that let you interact via REST instead of establishing and closing potentially expensive connections at each request.)

I don't know much about Firebase, but in these scenarios, you'd generally have to check if the app has already been initialized, and only execute ```initializeApp``` in case it hasn't been. But this is more related to Firebase and Serverless Arch. in general and not Next, so I suggest doing some research about using Firebase serverless.
